scipy/signal/_spectral_py.py

```diff
@@ -134,9 +134,9 @@ def lombscargle(x,
     >>> plt.show()
 
     """
-    x = np.asarray(x, dtype=np.float64)
-    y = np.asarray(y, dtype=np.float64)
-    freqs = np.asarray(freqs, dtype=np.float64)
+    x = np.ascontiguousarray(x, dtype=np.float64)
```

Comment:
I think this makes a copy for non-contiguous arrays. Is this a memory regression compared to the Cython version?

https://github.com/scipy/scipy/blob/2b526a9d2f0a85c804a2363e06111bd6976795e9/scipy/signal/_spectral.pyx#L21-L23

For the Cython version, there is no requirement for contiguous arrays.